### DISCUSSION TITLE
Host a bot as server

### DISCUSSION THREAD

**Discussion by noshila** [facebookresearch/ParlAI#3929] [General]
I want to run blender bot locally (would like to try both blender_90M as well as blenderbot2_400M) in browser, in a way that other people from computers (within the network i.e. intranet) can talk to it. What should I do?

**Comment by mojtaba-komeili** [comment]
If you want it to be accessible on the network/internet, you need to setup a light weight server that upon receiving a request from a new user creates an agent for the user to interact with and handles sending/receiving messages between the user and the model agent. You need to keep the connection alive, that means using something like a websocket (check [this](https://github.com/facebookresearch/ParlAI/blob/master/parlai/chat_service/services/websocket/agents.py))

**Reply by Tacacs-1101** [reply]
model_file = "zoo:blenderbot2/{}/model".format(config.model_name)   # point it to downloaded model dir if you have.

# parameters for blender2
opt = {
    "model": "projects.blenderbot2.agents.blenderbot2:BlenderBot2FidAgent",
    "model_file": model_file,
    "no_cuda": False,
    "override": {
        "no_cuda": False,
        "search_server": "127.0.0.1:10005",
        "history_size": 1,
    }
}

blender_agent = create_agent(opt)

I am creating agent like this for blenderbot2 globally for exposing it through grpc api.

passing a utterance like this
blender_agent.observe({'text': utt, "episode_done": False})
and getting prediction as blender_agent.act()

I want to expose this output through a grpc endpoint. I also want to reset history at each turn.
Is this the correct way, because my response are coming but that is not good enough as claimed by blenderbot.
